saic-java-mqtt-gateway/src/main/java/net/heberling/ismart/mqtt/VehicleState.java

```diff
@@ -455,17 +472,48 @@ public void notifyMessage(SaicMessage message) throws MqttException {
       msg.setRetained(true);
       client.publish(mqttVINPrefix + "/info/lastMessage", msg);
       lastVehicleMessage = message.getMessageTime();
+      resetApiUpdateError();
     }
     // something happened, better check the vehicle state
     notifyCarActivityTime(message.getMessageTime(), false);
   }
 
-  public boolean isRecentlyActive() {
-    return hvBatteryActive
-        || lastCarActivity.isAfter(ZonedDateTime.now().minus(15, ChronoUnit.MINUTES));
+  public boolean shouldRefresh() {
+    if (lastSuccessfulRefresh == null) {
+      markSuccessfulRefresh();
+      return true;
+    }
+    if (lastCarActivity.isAfter(lastSuccessfulRefresh)) {
+      return true;
+    }
+    if (apiUpdateError > 10) {
```

Comment:
If we get over 10 api errors, can they ever be reset?

Reply:
Yes, as soon as the car sends a message to the api or if before the 10 apiErrors it refreshes correctly. However, if the gateway goes into a non recoverable api error state, it will be doomed anyway... ;) Or do you have a better solution for this? Idea is that a failing api will not bomb the api.

One other idea is that the if somehow waits for like 1h if > apiErrors is > 10, this way a broken api will not be bombed with requests...

Reply:
I will remove this part for now. Not sure if it fits into this MR and is easily reimplemented, else we will be on this MR forever. :)